wrapanapi/systems/ec2.py

```diff
@@ -1595,3 +1596,41 @@ def cleanup_resources(self):
         self.remove_all_unused_nics()
         self.remove_all_unused_volumes()
         self.remove_all_unused_ips()
+
+    def list_ocps(self):
+        """
+        List openshift clusters (name, metadata, resources) where metadata can be used for uninstalling the cluster
+        """
+        ocp_list = []
+        resources = self.resourcegroupstaggingapi_connection.get_resources()["ResourceTagMappingList"]
+        for resource in resources:
+            tags = resource["Tags"]
+            for tag in tags:
+                key = tag["Key"]
+                if key.startswith("kubernetes.io/cluster/"):
```

Comment:
The GetResources endpoint supports `TagFilters`, and you can use a key regex expression to limit the response to only resources with this tag. Then you don't have to iterate so much here, and can directly fetch the objects that you want and parse their cluster key.

https://docs.aws.amazon.com/resourcegroupstagging/latest/APIReference/API_GetResources.html
https://docs.aws.amazon.com/resourcegroupstagging/latest/APIReference/API_TagFilter.html